src/main/java/org/jvnet/hudson/test/JellyTestSuiteBuilder.java

```diff
@@ -103,11 +111,52 @@ protected void runTest() throws Exception {
             if (requirePI) {
                 ProcessingInstruction pi = dom.processingInstruction("jelly");
                 if (pi == null || !pi.getText().contains("escape-by-default")) {
-                    throw new AssertionError("<?jelly escape-by-default='true'?> is missing in "+jelly);
+                    errors.add("<?jelly escape-by-default='true'?> is missing in "+jelly);
                 }
-
             }
             // TODO: what else can we check statically? use of taglibs?
+            checkScriptElement(dom);
+            checkJavaScriptAttributes(dom);
+            if (!errors.isEmpty()) {
+                String message = String.join("\n", errors);
+                throw new AssertionError(message);
+            }
+        }
+
+        private void checkJavaScriptAttributes(Document dom) {
+            List<Node> allNodes = dom.selectNodes("//*");
+            allNodes.forEach(n -> {
+                Element element = (Element) n;
+                Attribute onclick = element.attribute("onclick");
+                Attribute checkUrl = element.attribute("checkUrl");
+                Attribute checkDependsOn = element.attribute("checkDependsOn");
+                if (checkUrl != null  && checkDependsOn == null) {
+                    errors.add("Usage of 'checkUrl' without 'checkDependsOn' in "+jelly);
+                }
+                if (onclick != null && element.getNamespace() != Namespace.NO_NAMESPACE) {
+                    errors.add("Usage of 'onclick' from a taglib in "+jelly);
```

Comment:
could we link to a page explaining how to fix this?

Reply:
When used via a taglib it mean the resulting tag will also have that onclick handler. Afaik for all places in core that define an `onclick` in a taglib it is marked as deprecated. So the usage of onclick from a taglib is not really different from an onclick on a plain html element.